alert/log/store.go

```diff
@@ -637,7 +644,8 @@ func (db *DB) Search(ctx context.Context, opts *SearchOptions) ([]Entry, int, er
 	(coalesce(a.timestamp < cast($4 as timestamp with time zone), true)) and
 	($5 = '' or a.event = $5::enum_alert_log_event)and
 	($6 = '' or a.sub_user_id = cast($6 as UUID)) and 
-	($7 = '' or a.sub_integration_key_id = cast($7 as UUID))`
+	($7 = '' or a.sub_integration_key_id = cast($7 as UUID)) and
+	($8 = '0' OR a.id < $8 :: int)`
```

Comment:
I think we should revert this method to the original, and rename it to `LegacySearch` as we've done elsewhere (when switching to cursor pagination) and then implement a new `Search` method for cursor pagination.

This one is using count + limit/offset and wont give what we are expecting if we combine the two, or in the least will make it more confusing to follow.